addons/im_livechat/models/mail_channel.py

```diff
@@ -191,3 +194,134 @@ def _email_livechat_transcript(self, email):
+            'question_email': 'email',
+            'question_phone': 'phone',
+        })
+        return user_input.get('email', False), user_input.get('phone', False)
+
+    def _chatbot_post_message(self, chatbot, body, text_to_html=True):
+        if text_to_html:
+            body = plaintext2html(body)
+
+        return self.with_context(mail_create_nosubscribe=True).message_post(
+            author_id=chatbot.operator_partner_id.id,
+            body=body,
+            message_type='comment',
+            subtype_xmlid='mail.mt_comment',
+        )
+
+    def _chatbot_process_step(self, step_id):
+        self.ensure_one()
+        if self.livechat_chatbot_current_step_id and step_id.sequence <= self.livechat_chatbot_current_step_id.sequence:
+            raise ValidationError(_('The new step sequence must be higher than the old one.'))
+
+        # We change the current step to the new step
+        self.livechat_chatbot_current_step_id = step_id
+
+        human_operator = False
+        if step_id.step_type == 'forward_operator':
+            human_operator = self.livechat_channel_id._get_random_operator() if self.livechat_channel_id else False
+            if human_operator:
+                self.with_user(self.env.ref('base.user_root')).add_members(
+                    human_operator.partner_id.ids,
+                    open_chat_window=True,
+                    post_joined_message=False)
+
+                if step_id.message:
+                    # first post the message of the step (if we have one)
+                    posted_message = self._chatbot_post_message(step_id.chatbot_id, step_id.message)
+                else:
+                    posted_message = False
+
+                # then post a small custom 'Operator has joined' notification
+                self._chatbot_post_message(
+                    step_id.chatbot_id,
+                    '<div class="o_mail_notification">%s</div>' % _('%s has joined', human_operator.partner_id.name),
+                    text_to_html=False)
+            else:
+                # don't post anything as the bot in case we did not find an operator
+                # the script will continue normally
+                posted_message = False
+        else:
+            posted_message = self._chatbot_post_message(step_id.chatbot_id, step_id.message)
+
+        if human_operator:
+            self._broadcast(human_operator.partner_id.ids)
+            self.channel_pin(self.uuid, pinned=True)
+
+        return posted_message
+
+    def _chatbot_validate_email(self, email_address, chatbot):
+        email_normalized = email_normalize(email_address)
+
+        posted_message = False
+        if not email_normalized:
+            posted_message = self.with_context(mail_create_nosubscribe=True).message_post(
+                author_id=chatbot.operator_partner_id.id,
+                body=_('This does not look like a valid email, can you please try again?'),
```

Comment:
Maybe "%(input_email)s does not look like a valid email. Can you please try again?", to include the given input ?